src/model/MultiValueColumn.ts

```diff
@@ -86,151 +89,130 @@ export default class MultiValueColumn extends ValueColumn<number[] > {
 
   constructor(id: string, desc: any) {
     super(id, desc);
-    this.colorrange = (<any>desc).colorrange || ['blue', 'red'];
+    this.colorRange = (<any>desc).colorRange || ['blue', 'red'];
     this.min = d3.min((<any>desc).domain);
     this.max = d3.max((<any>desc).domain);
-    this.bins = (<any>desc).datalength;
+    this.dataLength = (<any>desc).dataLength;
     this.threshold = (<any>desc).threshold || 0;
-
     this.sortBy = (<any>desc.sort) || 'min';
-    this.verticalBarHeight = 13;
-    this.rendererList = [{type: 'heatmapcustom', label: 'Heatmap'},
+    this.cellWidth = this.getWidth() || 100;
+    this.rowHeight = 13;
+    this.dataInfo = {
+      min: this.min,
+      max: this.max,
+      threshold: this.threshold,
+      sort: this.sortBy
+    };
+    this.rendererInfo.rendererList = [{type: 'heatmapcustom', label: 'Heatmap'},
       {type: 'boxplot', label: 'Boxplot'},
       {type: 'sparkline', label: 'Sparkline'},
       {type: 'threshold', label: 'Threshold'},
       {type: 'verticalbar', label: 'VerticalBar'}];
 
-    this.defineColor();
-    this.boxPlotWidth();
+    this.defineColorRange();
 
   }
 
-  private defineColor() {
+  private defineColorRange() {
 
     if (this.min < 0) {
       this.colorScale
         .domain([this.min, 0, this.max])
-        .range([this.colorrange[0], 'white', this.colorrange[1]]);
+        .range([this.colorRange[0], 'white', this.colorRange[1]]);
 
     } else {
       this.colorScale
         .domain([this.min, this.max])
-        .range(['white', this.colorrange[1]]);
+        .range(['white', this.colorRange[1]]);
     }
   }
 
 
-  private boxPlotWidth() {
-    this.boxPlotScale
-      .domain([this.min, this.max])
-      .range([0, this.getWidth()])
-
-  }
-
-
   compare(a: any, b: any, aIndex: number, bIndex: number) {
-    this.sortBy = this.getUserSortBy();
-    const a_val = this.getValue(a, aIndex);
-    const b_val = this.getValue(b, bIndex);
+
+    const a_val = (this.getValue(a, aIndex)).sort(numSort);
+    const b_val = (this.getValue(b, bIndex)).sort(numSort);
     const sort: any = new CustomSortCalculation(a_val, b_val);
     const f = sort[this.sortBy].bind(sort);
 
     return f();
   }
 
+  getColor() {
 
-  getLabel(row: any, index: number) {
-    return '' + this.getValue(row, index);
-  }
-
-  getRaw(row: any, index: number) {
-    return this.accessor(row, index, this.id, this.desc, this.findMyRanker());
-  }
-
-  getValue(row: any, index: number) {
-    var v = this.getRaw(row, index);
-    return (v);
+    return this.colorScale;
   }
 
-  getColor(data: any) {
+  calculateCellDimension(width) {
 
-    return this.colorScale(data);
-  }
-
-  calculateCellDimension() {
 
-    return (this.getWidth() / this.bins);
+    return (width * 1 / this.dataLength);
   }
 
-  getxScale(data) {
+  getSparkLineXScale() {
     this.xposScale
```

Comment:
isn't `this.dataLength` constant and just set once within the constructor? if so, there is no need to update the scale on each getter call